cuda_core/cuda/core/experimental/_memory.py

```diff
@@ -286,3 +286,34 @@ def is_host_accessible(self) -> bool:
     @property
     def device_id(self) -> int:
         raise RuntimeError("the pinned memory resource is not bound to any GPU")
+
+
+class _AsyncMemoryResource(MemoryResource):
+    __slots__ = ("_dev_id",)
+
+    def __init__(self, dev_id):
+        self._handle = None
+        self._dev_id = dev_id
+
+    def allocate(self, size, stream=None) -> Buffer:
+        if stream is None:
+            stream = default_stream()
+        ptr = handle_return(cuda.cuMemAllocAsync(size, stream._handle))
```

Comment:
The whole point of the TCC support is to not use the stream ordered memory allocator. This API would still require the memory pool support (it draws memory from the "current" (or "default" if no pool is set current) pool), which would not work on TCC. We need to just use cuMemAlloc here.

Reply:
ack, I will switch it over. For my own understanding: When I tested, I got a false positive because the allocator drew memory from the default pool whcih is supported on TCC, but if I were to have set the pool to something other than default, it would have failed. Is that correct?